default.config.js

```diff
@@ -155,6 +155,7 @@ module.exports = {
   },
 
   server: {
+    vaultSeed: '764c1afa2988981a',
```

Comment:
I'm not at ease with that.
Instead, I think you should do as with the JWT secret key: if it's not null in the configuration file, then use that one. Otherwise, create a fresh new key and store it in the database.

You can use the JWT key generator as an example: https://github.com/kuzzleio/kuzzle/blob/master/lib/services/internalEngine/bootstrap.js#L225

Reply:
That what I wanted to do in the first time but I also have to use the Vault in the CLI, without access to internal database and without generation of a key like for the JWT secret.  

Moreover the generated key changes when you restart Kuzzle (with docker-compose down) or wipe the internal database and the secrets can not be decrypted any more since the IV is not the same.   

This is why the initialization vector must be set in the configuration file and not auto generated. (I added warning messages to indicate developers to do so and I will also insist on that point in the documentation)

Reply:
About your 1st point: you can either use the CLI as an API client as with the other commands, or maybe you can bootstrap our internal engine client?

Your second point is moot, since:
1. the proposed solution still allows setting a key in the configuration files
2. it's a good thing that, after you destroy completely a kuzzle instance, a fresh one is unable to decrypt the vault without the right seed. That's what the vault is for, isn't it?

Reply:
You propose to add methods to the admin controller ? I thought about it but I feel it overkill.  

About my second point, you're right on 1) and 2). But I was thinking that maybe the better solution is not to generate the key ourself and let the developers manually set it in the kuzzlerc file. 
If they does not put a value to `server.vaultKey`, the Vault is just deactivated

Reply:
I agree about how you treated the key, and especially about deactivating the vault if no key was set. :+1: 
I was of course only talking about the seed.

Reply:
Hmm I think I misuse some words..

I was talking about the seed, deactivate the Vault if the developers does not have set a value for `server.vaultSeed`

Reply:
I think that only the presence of a key should tell Kuzzle to (de)activate the vault. The seed in itself has no meaning, it can only concomitantly to a key.

But what we can do is to ignore any seed value if there isn't a key, and only auto-generate a seed if a key is provided without a seed.

Reply:
I can auto-generate a seed but either I have to put it in the kuzzlerc in the same time or the developer has to do it manually.  
If the seed is stored only in the database, other developers cloning the project will not have the same value for the seed

Reply:
Ah, right, I get it: auto-generating a seed binds the encrypted file to a Kuzzle instance. 
We could add a small command printing the current seed, though, but that doesn't seem practical.

And I can't shake the feeling that proposing a default seed doesn't look good at all.

New idea then: what about using the key as a seed (stretching it if needed) if none is provided?
Of course, we would have to keep that big warning you added (we would have to reword it), but in essence, that would mean the following: "adding a seed is optional, but strongly encouraging to reinforce encryption"
And we could do that without having a "default seed" (which means having no seed at all)

What do you think?

Reply:
I like this proposal ;)